src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -971,5 +974,15 @@ private static LSP.Range MappedSpanResultToRange(MappedSpanResult mappedSpanResu
                 End = LinePositionToPosition(mappedSpanResult.LinePositionSpan.End)
             };
         }
+
+        /// <summary>
+        /// Retrieves the <see cref="LSP.TextDocumentIdentifier"/> from the request data.
+        /// </summary>
+        public static LSP.TextDocumentIdentifier? GetTextDocument(object? requestData)
+        {
+            Contract.ThrowIfNull(requestData);
+            var resolveData = ((JToken)requestData).ToObject<DocumentResolveData>();
+            return resolveData?.TextDocument;
+        }
```

Comment:
i don't love this method.  IT seems to implicitly depend on the value being passed in being convertible to a DocumentResultData.  can this be inlined to whoever needs to call this?

Reply:
Yeah, there's only one left now